### PR TITLE
fix: refactor code and fix progress null after retry bug

### PR DESCRIPTION
**Pull request by ia3andy**
Fixes https://github.com/fabric8-launcher/launcher-frontend/issues/663

### CODE REVIEW

**Comment by ia3andy** [comment]
You can now view the retry working in the demo

File: projects/ngx-launcher/src/lib/create-app/project-progress-createapp-nextstep/project-progress-createapp-nextstep.component.ts [hunk 1]
```diff
@@ -86,10 +77,15 @@ export class ProjectProgressCreateappNextstepComponent implements OnChanges, OnD
...
+      this.launcherComponent.summary, failedStep).subscribe(val => {
         this.errorMessage = null;
-        this.ngOnChanges({'statusLink': new SimpleChange('', result.uuid_link, false)});
+        if (!val || !val['uuid_link']) {
+          this.errorMessage = 'Invalid response from server!';
+        }
+        for (const step of this._progress.filter((s) => !s.completed)) {
+          step.error = false;
+        }
+        this.launcherComponent.statusLink = val['uuid_link'];
```
Comment (edewit) [inline projects/ngx-launcher/src/lib/create-app/project-progress-createapp-nextstep/project-progress-createapp-nextstep.component.ts]: will the `ngOnChanges` be called when you only do this?

Comment (ia3andy) [reply projects/ngx-launcher/src/lib/create-app/project-progress-createapp-nextstep/project-progress-createapp-nextstep.component.ts]: @edewit it does, you can try in the demo

**Pull request merged**